### ISSUE TITLE
bv.get_code_refs instantiates new architecture for every BNReferenceSource

### ISSUE THREAD

**Issue by bambu** [Vector35/binaryninja-api#986]
`bv.get_code_refs` instantiates a `architecture.CoreArchitecture` for every architecture in each `BNReferenceSource`. This causes a symbol with just over 2000 code_refs to take roughly 10 seconds.

```python
>>> # Example no caching:
>>> start = time.time(); refs = bv.get_code_refs(new.address); print time.time() - start
10.9354510307
>>> len(refs)
2191
```

Caching the architecture and retrieving it if it already exists dramatically increases the speed.

```python
>>> # Example caching:
>>> start = time.time(); refs = bv.get_code_refs(new.address); print time.time() - start
0.0381011962891
>>> len(refs)
2191
```

**Comment by joshwatson** [comment]
I thought arch was already cached now? Or was that something else that was cached in a previous issue?

**Comment by bambu** [comment]
I don't see any caching so I'm assuming it was something else. Even looking up the architectures as a list or dictionary is not cached and get recreated every time. So if you are doing `Architecture['arch']` in a loop it will create a new `CoreArchitecture` for each result.

**Comment by plafosse** [comment]
The architecture was cached for some APIs, but @rssor has a change he'll be pushing soon with a new caching method which will cache the handle for each CoreArchitecture instance.

**Comment by rssor** [comment]
This has been fixed in build 1.1.1125-dev.
